web-app/packages/lib/src/modules/project/types.ts

```diff
@@ -285,10 +285,20 @@ export interface ProjectTemplate {
 export type EnhancedProjectDetail = ProjectDetail & {
   files: Record<string, ProjectListItemFiles>
   path: string
-  versions?: ProjectVersion[]
 }
 
 export interface UpdateProjectAccessParams {
   user_id: number
   role: ProjectRoleName
 }
+
+export interface FetchProjectVersionsPayload extends ProjectParams {
+  params: FetchProjectVersionsParams
+}
+
+export interface ProjectVersionsPayload {
+  versions: ProjectVersion[]
+  count: number
+}
+
+export type ProjectVersionsItem = ProjectVersion & { disabled: boolean }
```

Comment:
Is there some reason why you prefer type intersection ('&') here over extending interface?

(similarily few lines above for type EnhancedProjectDetail)

Reply:
Sometimes type, sometimes interface :). Disabled is not occuring in ProjectVersion, so its safe solution. But, maybe extends is better here :)